### PR TITLE
Dispose of WebSocket when done with it

### PR DESCRIPTION
**Pull request by Shane32**

### CODE REVIEW

**Comment by STRATZ-Ken** [comment]
Can you push this to a preview build so I can test on production?

**Comment by Shane32** [comment]
@sungam3r Can you review this quickly?

**Comment by Shane32** [comment]
@sungam3r Merging now so @STRATZ-Ken can test it.  Please comment when you get a chance.  The API does change, but it's a non-breaking change.

**Comment by Shane32** [comment]
Published as nuget version `5.1.0-preview-127`

File: src/Transports.Subscriptions.WebSockets/WebSocketConnection.cs [hunk 1]
```diff
@@ -18,9 +18,16 @@ public class WebSocketConnection
...
-            await _transport.CloseAsync();
+            try
+            {
+                await _server.OnConnect();
+                await _server.OnDisconnect();
+                await _transport.CloseAsync();
+            }
+            finally
+            {
+                _transport.Dispose();
```
Comment (sungam3r) [inline src/Transports.Subscriptions.WebSockets/WebSocketConnection.cs:29]: Dispose transport? Why?

Comment (sungam3r) [reply src/Transports.Subscriptions.WebSockets/WebSocketConnection.cs:29]: `WebSocketConnection` should implement `IDisposable` and dispose `_transport`.

Comment (sungam3r) [reply src/Transports.Subscriptions.WebSockets/WebSocketConnection.cs:29]: Even CodeQL shows warning above.

**Comment by sungam3r** [comment]
Reviewed.

File: src/Transports.Subscriptions.WebSockets/WebSocketConnection.cs (hunk 1 above)
Comment (Shane32) [reply src/Transports.Subscriptions.WebSockets/WebSocketConnection.cs:29]: Already in PR #673 but if the code doesn't work now, it won't work with that change.

**Pull request merged**